.gitignore

```diff
@@ -0,0 +1,2 @@
+*.gcode
+**__pycache__**
```

Comment:
> I think we need to add [at a minimum] some gcode files to serve as _before_ and _after_ examples to be used to test code refactors and optimizations. But eventually you are going to need full blown unit tests.

Agreed that would be useful. I made my own while I was refactoring so I could know the results weren't just garbage. The only change that actually affects the generated output is the rounding in the `writeLine` function, because:
- it's now f-string style, which includes trailing zeros (e.g. `str(round(1.23999, 3))` -> `'1.24'`, whereas `f'{1.23999:.3f}'` -> `'1.240'`
- it now supports user-specified precision, which can be used to reduce file sizes for less accurate printers/prints

> I think you should also include `*.pyc`

I believe they're always generated inside a `__pycache__` directory?